### PR TITLE
Port flexible_event_privacy.py to TypeScript

### PR DESCRIPTION
**Pull request by apasel422**
This will allow us to incorporate this information into the header validator.

### CODE REVIEW

File: flexible-event/index.js [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+  console.log(`Information gain: ${infoGain} bits`)
+  console.log(`Flip percent: ${100 * flipProb}%`)
+
+  const infoGainDefault = maxInformationGain(
+        numFlexibleStates(DEFAULT_CONFIG[sourceType]), epsilon)
+
+  if (infoGain > infoGainDefault) {
+      const [newEps, flipProb] = epsilonToBoundInfoGainAndDp(numStates, infoGainDefault, epsilon)
+      console.log(
+          `WARNING: info gain of ${infoGain} > ${infoGainDefault} for ${sourceType} sources. Would require a ${100 * flipProb}% flip chance (effective epsilon = ${newEps}) to resolve.`)
```
Comment (csharrison) [inline flexible-event/index.js:154]: The output is a lot nicer to read if these are capped to a certain # of decimal points using e.g. `toFixed(<decimal places>)`

File: flexible-event/index.js [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
+  return {maxEventLevelReports, perTriggerDataConfigs}
+}
+
+const optionDefs = [
+  {
+    name: 'max_event_level_reports',
+    alias: 'm',
+    type: Number,
+    defaultValue: 20,
```
Comment (csharrison) [inline flexible-event/index.js:206]: The default should be based on the source type

**Review by csharrison — CHANGES_REQUESTED** [review]

File: flexible-event/index.js [hunk 3]
```diff
@@ -0,0 +1,253 @@
+const commandLineArgs = require('command-line-args')
```
[the remark this answers is not in the document]
Comment (apasel422) [reply flexible-event/index.js:1]: That would be fine with me, but @maudnals configured our other JS code to not use them.

File: flexible-event/index.js (hunk 2 above)
Comment (apasel422) [reply flexible-event/index.js:206]: The original code used 20, no?

Comment (csharrison) [reply flexible-event/index.js:208]: Ah yeah, sorry this is fine.

File: flexible-event/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (csharrison) [reply flexible-event/index.js:1]: ack, that's fine.

**Review by csharrison — APPROVED** [review]

**Review by csharrison — APPROVED** [review]

**Pull request merged**